lesson3

KochevniKXXI opened the pull request.

KochevniKXXI committed:
Третий урок

DmitriyTSt commented on Lesson_3_Zemskov/app/src/main/java/hh/school/lesson_3_zemskov/ProfileActivityOnConstraintLayout.kt:
```diff
@@ -0,0 +1,73 @@
+        }
+
+        with(binding) {
+            val personalDataLabels = resources.getStringArray(R.array.personal_data_labels)
```
не понял такой логики со списком, логичнее под каждый лейбл строковый ресурс завести, так ведь вообще неочевидно

DmitriyTSt commented on Lesson_3_Zemskov/app/src/main/java/hh/school/lesson_3_zemskov/ProfileActivityOnConstraintLayout.kt:
```diff
@@ -0,0 +1,73 @@
+        binding = ActivityProfileOnConstraintLayoutBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        setSupportActionBar(binding.toolbar)
```
давай без supportActionBar

я на уроке показывал
setNavigationOnClickListener
setOnMenuItemClickListener

DmitriyTSt commented on Lesson_3_Zemskov/app/src/main/java/hh/school/lesson_3_zemskov/ProfileActivityOnConstraintLayout.kt:
```diff
@@ -0,0 +1,73 @@
+        }
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
```
оверайды про меню тоже уйдут

DmitriyTSt commented on Lesson_3_Zemskov/app/src/main/java/hh/school/lesson_3_zemskov/ProfileActivityWithoutConstraintLayout.kt:
```diff
@@ -0,0 +1,73 @@
+import hh.school.lesson_3_zemskov.databinding.ActivityProfileWithoutConstraintLayoutBinding
+import hh.school.lesson_3_zemskov.model.User
+
+class ProfileActivityWithoutConstraintLayout : AppCompatActivity() {
```
тут те же замечания что в первой активити

DmitriyTSt commented on Lesson_3_Zemskov/app/src/main/res/layout/activity_profile_on_constraint_layout.xml:
```diff
@@ -0,0 +1,100 @@
+            <include
+                android:id="@+id/textViewHeader"
+                layout="@layout/app_bar_text_view_header"
+                android:layout_width="wrap_content"
```
лучше заботиться о длинных данных (так как они сетятся из кода), и проверять на длинных строках, и привязывать к концу тоже, чтобы отступ справа предусмотреть

DmitriyTSt commented on Lesson_3_Zemskov/app/src/main/res/layout/activity_profile_on_constraint_layout.xml:
```diff
@@ -0,0 +1,100 @@
+
+    <ScrollView
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
```
кроме скролл вью ничего не будет, логичнее сделать высоту 0dp и привязать к низ родителя

DmitriyTSt commented on Lesson_3_Zemskov/app/src/main/res/layout/activity_profile_without_constraint_layout.xml:
```diff
@@ -0,0 +1,99 @@
+                <include
+                    android:id="@+id/textViewHeader"
+                    layout="@layout/app_bar_text_view_header"
+                    android:layout_width="wrap_content"
```
тут тоже замечание, что лучше на всю ширину дабы предусмотреть отступы справа

DmitriyTSt commented on Lesson_3_Zemskov/app/src/main/res/layout/activity_profile_without_constraint_layout.xml:
```diff
@@ -0,0 +1,99 @@
+
+    <androidx.core.widget.NestedScrollView
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
```
match_parent, а еще тогда коллапсинг будет работать за сам нестед скрол, а не только за шапку

DmitriyTSt commented on Lesson_3_Zemskov/app/src/main/res/layout/activity_profile_without_constraint_layout.xml:
```diff
@@ -0,0 +1,99 @@
+
+                <include
+                    android:id="@+id/listItemExitAccount"
+                    layout="@layout/list_item" />
```
а зачем кнопку выхода так обобщать? она вообще никак на параметры не похожа, логичнее сделать отдельно

KochevniKXXI replied:
исключительно в учебных целях решил попробовать сделать такой переиспользуемый элемент. Переделать?

KochevniKXXI committed:
Третий урок (ред.)
1. Строковый массив заголовков полей разделён на строковые ресурсы.
2. Исключено использование supportActionBar и override методов для меню. Вместо этого:
setNavigationOnClickListener
setOnMenuItemClickListener.
3. Исправлены параметры ширины, либо привязок текстов и прокруток.
